app/Models/Transaction.php

```diff
@@ -220,6 +231,15 @@ public function vendorField(): string|null
         return $this->vendorFieldContent;
     }
 
+    public function fee(): BigNumber
+    {
+        if ($this->receipt === null) {
+            return $this->gas_price;
+        }
+
+        return $this->gas_price->multipliedBy($this->receipt->gas_used->valueOf());
```

Comment:
this will be in gwei (10^-9) so by default I'd suggest we show it in terms of ARK by dividing it by that value. Will also be helpful to add helpers to switch between ARK / gwei / wei values 🤔